### DISCUSSION TITLE
How do I debug `the specifiers must be a non-empty string. Receive ""`

### DISCUSSION THREAD

**Discussion by MIreland** [web-infra-dev/rsbuild#3402] [Q&A]
Hi,

I'm upgrading from 0.6.15 to 1.0.1-rc.5

I'm receiving the following error:

```
Failed to compile, check the errors for troubleshooting.
File: /Users/user/Desktop/orthlyweb/node_modules/.pnpm/assert@2.0.0/node_modules/assert/build/assert.js:1:1
  × The specifiers must be a non-empty string. Received ""

File: /Users/user/Desktop/orthlyweb/node_modules/.pnpm/assert@2.0.0/node_modules/assert/build/internal/assert/assertion_error.js:1:1
  × The specifiers must be a non-empty string. Received ""

File: /Users/user/Desktop/orthlyweb/node_modules/.pnpm/assert@2.0.0/node_modules/assert/build/internal/errors.js:1:1
  × The specifiers must be a non-empty string. Received ""

File: /Users/user/Desktop/orthlyweb/node_modules/.pnpm/assert@2.0.0/node_modules/assert/build/internal/util/comparisons.js:1:1
  × The specifiers must be a non-empty string. Received ""
```

<img width="1086" alt="image" src="https://github.com/user-attachments/assets/f27e3f22-6708-4c75-852a-5bbd5b35959a">

How do I go about debugging this? I've upgraded the config according to what I see in [the upgrade guide.](https://github.com/web-infra-dev/rsbuild/discussions/2508).

Thanks!

**Comment by MIreland** [comment]
We had a number of aliases to polyfill node packages. Those appear to have been the issue, so we can move forward on our end.

I'll leave this open for now in hopes of surfacing a better way to debug this.

**Reply by chenjiahan** [reply]
Are you using this plugin? https://github.com/rspack-contrib/rsbuild-plugin-node-polyfill

**Reply by MIreland** [reply]
We were. We also had a number of manually created aliases pointing to browserify polyfills. By removing those aliases, it was no longer crashing.
